File: DMOJ Submissions/sssp.py
n,m = map(int,input().split())

graph = [[] for i in range(n+1)]

for i in range(m):
    a,b,w = map(int,input().split())

    graph[a].append([b,w])
    graph[b].append([a,w])

inf = 1e9

dist = [inf for i in range(n+1)]
inq = [0 for i in range(n+1)]
q = [1]

dist[1] = 0
inq[1] = 1

while(q):
    at = q.pop(0)
    inq[at] = 0

    for i in graph[at]:
        if dist[at] + i[1] < dist[i[0]]:
            dist[i[0]] = dist[at] + i[1]
            if not inq[i[0]]:
                q.append(i[0])
                inq[i[0]] = 1

for i in range(1,n+1):
    if dist[i] == inf:
        print(-1)
    else:
        print(dist[i])

File: DMOJ Submissions/sssp.cpp
#include <bits/stdc++.h>

using namespace std;

#define MAXN numeric_limits<int>::max()
vector<vector<pair<int,int> > > graph;
vector<int> processed;
vector<int> dist;

int main(){
	int n,m,x,y,w;
	cin >> n >> m;
	vector<pair<int,int> > temp;
	for(int i = 0; i <= n; i++){
		graph.push_back(temp);
		dist.push_back(MAXN);
		processed.push_back(false);
	}
	dist[1] = 0;
	for(int i = 0; i < m; i++){
		scanf("%i %i %i",&x,&y,&w);
		pair<int,int> t (x,w);
		pair<int,int> t2 (y,w);
		graph[x].push_back(t2); graph[y].push_back(t);
	}
	priority_queue<pair<int,int> > q;
	pair<int,int> start(0,1);
	q.push(start);
	while(!q.empty()){
		int a = q.top().second; q.pop();
		if(processed[a]) continue;
		for(auto u: graph[a]){
			int b = u.first; int w = u.second;
			if(w+dist[a]<dist[b]){
				dist[b] = w + dist[a];
				pair<int,int> add (-dist[b],b);
				q.push(add);
			}
		}
	}
	for(int i = 1; i <= n; i++){
		if(dist[i]==MAXN){
			cout << "-1\n";
		}
		else{
			cout << dist[i] << "\n";
		}
	}
}